### PR TITLE
mrc-785 Show session expired message in front end

### PR DESCRIPTION
**Pull request by EmmaLRussell**
This is a bit of a hack as I've put the actual error message into the front end rather than json response from the backend, where it was relatively simple to return an empty 401 response rather than redirecting, but unless I'm missing something really obvious, tricky to put in a JSON response entity. Theoretically could override SecurityInterceptor and hook into the HttpServletResponse object (same one the Filter uses), but I was struggling to get that to honour the new content in the response emitted through the response wrapper (although it would get logged). The message is kind of specific to the front end anyway... 

You can test this manually without having to hang around for an hour by adding "server.servlet.session.timeout=60s" to application.properties.

### CODE REVIEW

**Comment by EmmaLRussell** [comment]
Have created ticket for not attempting validate/delete if get a 401. (assume it's just on 401?)
https://vimc.myjetbrains.com/youtrack/issue/mrc-1180

File: src/app/src/main/kotlin/org/imperial/mrc/hint/AppStart.kt [hunk 1]
```diff
@@ -29,9 +29,14 @@ class MvcConfig(val config: Config) : WebMvcConfigurer {
     }
 
     override fun addInterceptors(registry: InterceptorRegistry) {
+        //Root url - redirect to login if required
         registry.addInterceptor(SecurityInterceptor(config, "FormClient"))
+                .addPathPatterns("/")
+
+        //Ajax endpoints - secure, but do not redirect with login form client - return a 401
+        registry.addInterceptor(SecurityInterceptor(config, ""))
```
[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply src/app/src/main/kotlin/org/imperial/mrc/hint/AppStart.kt]: yeah, hence the comment!

**Comment by hillalex** [comment]
> Have created ticket for not attempting validate/delete if get a 401. (assume it's just on 401?)
> https://vimc.myjetbrains.com/youtrack/issue/mrc-1180

Oops, I did too. Working on it now. I'll close one as a duplicate.

**Comment by EmmaLRussell** [comment]
It seemed to work better to exclude /logout from both clients - otherwise I was getting redirected to login twice (login once, then see login screen again..)

**Pull request merged**